### PR TITLE
Fixing iOS Submission

### PR DESCRIPTION
**Pull request by Twolewis**
Fixes iOS Submission issues.  By and large bumping iOS to 1.11.344 of the AWS SDK

### CODE REVIEW

File: Gems/AWSCore/Code/Include/Framework/AWSApiClientJobConfig.h [hunk 1]
```diff
@@ -10,6 +10,7 @@
 
 #include <AzCore/std/smart_ptr/make_shared.h>
 #include <Framework/AWSApiJobConfig.h>
+#include <aws/s3/S3Client.h>
 
 // Forward declarations
```
Comment (Twolewis) [inline Gems/AWSCore/Code/Include/Framework/AWSApiClientJobConfig.h:15]: Cherrypicked from o3de/development

File: cmake/3rdParty/Platform/iOS/BuiltInPackages_ios.cmake [hunk 2]
```diff
@@ -19,7 +19,7 @@ ly_associate_package(PACKAGE_NAME glad-2.0.0-beta-rev2-multiplatform TARGETS gla
 ly_associate_package(PACKAGE_NAME expat-2.4.2-rev2-ios           TARGETS expat           PACKAGE_HASH 7c2635422aec23610f1633d5ed69189eb7704647133e009276c5dc566b5dedc1)
 ly_associate_package(PACKAGE_NAME tiff-4.2.0.15-rev3-ios         TARGETS TIFF            PACKAGE_HASH e9067e88649fb6e93a926d9ed38621a9fae360a2e6f6eb24ebca63c1bc7761ea)
 ly_associate_package(PACKAGE_NAME freetype-2.11.1-rev1-ios       TARGETS Freetype        PACKAGE_HASH b619eaa7da16469ae455a6418fbcd7bae8be406b69a46f5bc1706b275fb3558f)
-ly_associate_package(PACKAGE_NAME AWSNativeSDK-1.9.50-rev1-ios   TARGETS AWSNativeSDK    PACKAGE_HASH c3c9478c259ecb569fb2ce6fcfa733647adc3b6bd2854e8eff9de64bcd18c745)
+ly_associate_package(PACKAGE_NAME AWSNativeSDK-1.11.344-rev1-ios TARGETS AWSNativeSDK    PACKAGE_HASH 682325c035fa20b850ae6951b328560509834350d9ebc3e709284233f380b60b)
```
Comment (Twolewis) [inline cmake/3rdParty/Platform/iOS/BuiltInPackages_ios.cmake:22]: 1.11.344 is a custom package, not yet available on O3DE.

File: Gems/AWSCore/Code/Include/Framework/AWSApiClientJobConfig.h [hunk 3]
```diff
@@ -98,26 +99,54 @@ namespace AWSCore
...
         std::shared_ptr<ClientType> m_client;
     };
+
+    template<class ClientType>
+    std::shared_ptr<ClientType> AwsApiClientJobConfig<ClientType>::CreateClient() const
+    {
+        auto provider = GetCredentialsProvider();
+        if (provider != nullptr)
+        {
+            // Note: This constructor for AWS Client is marked for depreciation
```
Comment (carbonated-niko) [inline Gems/AWSCore/Code/Include/Framework/AWSApiClientJobConfig.h:119]: damn nothing holds value these days (depreciation)

Comment (carbonated-niko) [reply Gems/AWSCore/Code/Include/Framework/AWSApiClientJobConfig.h:119]: is this code also from o3de dev?

**Review by carbonated-niko — APPROVED** [review]

**Pull request merged**